.github/workflows/test.yaml

```diff
@@ -39,13 +42,20 @@ jobs:
       - name: Environment setup
         uses: ./.github/actions/setup
 
+      - name: Login to Registry1
+        uses: docker/login-action@v3
+        with:
+          registry: registry1.dso.mil
+          username: ${{ secrets.IRON_BANK_ROBOT_USERNAME }}
+          password: ${{ secrets.IRON_BANK_ROBOT_PASSWORD }}
+
       - name: Test a single source package
-        if: ${{ inputs.package != 'all' }}
-        run: uds run test-single-package
+        if: ${{ inputs.package != 'all' && matrix.flavor == 'upstream'}}
```

Comment:
Don't have a _strong_ opinion here but I wonder if it's more valuable to switch this to use registry1 for the single package tests (instead of upstream). My thinking here is that people are more likely to be doing upstream tests locally (and those are more likely to "just work" since its all upstream), so those pieces are generally going to be tested before being pushed. With that in mind...short circuiting CI on registry1 might make more sense?

Reply:
I do agree we are more likely to catch an issue on the registry1 tests. What is in here now is what we had agreed upon previously. 
I'm inclined to possibly just do both since the single capability tests are on public (free) runners and will still provide the short circuit when needed, with slightly more feedback, slightly less CI logic and no expense.

Reply:
we're doing both after the workflow refactor.